inngest/src/main/kotlin/com/inngest/Function.kt

```diff
@@ -255,4 +259,23 @@ internal open class InternalInngestFunction(
         // TODO use URL objects for serveUrl instead of strings so we can fetch things like scheme
         return configBuilder.build(client.appId, serveUrl)
     }
+
+    private fun serializeStepData(stepData: Any?): JsonNode? {
+        if (stepData == null) {
+            return stepData
+        }
+
+        val mapper = ObjectMapper()
+        val jsonString = mapper.writeValueAsString(stepData)
+        val readOnlyJson = mapper.readTree(jsonString)
```

Comment:
converting an object to a JSON string only to read that back in to get a JSON node is silly, but replacing this with

`val writeableJson = mapper.convertValue(stepData, ObjectNode::class.java)`

resulted in a bunch of test failures, so we decided to punt on figuring this out to a follow up

Reply:
Sounds good - I think I may have had code like this in an early PoC. There wasn't a great pattern for this in Java and the JSON libraries from what I could tell that didn't result in duplicate parsing/serialization work. I'm find with leaving this for a future improvement.